warehouse/models/mart/gtfs/fct_vehicle_locations.sql

```diff
@@ -0,0 +1,41 @@
+
+WITH
+
+-- base64_url
+-- header_timestamp
+-- vehicle_timestamp
+-- vehicle_id
+-- trip_id
+-- is unique
+fct_vehicle_positions_messages AS (
+    SELECT * FROM {{ ref('fct_vehicle_positions_messages') }}
+)
+
+, coalesced_and_filtered AS (
+    SELECT * EXCEPT (key)
+        , COALESCE(vehicle_timestamp, header_timestamp) AS location_timestamp
+    FROM fct_vehicle_positions_messages
+    WHERE trip_id IS NOT NULL
+        AND _gtfs_dataset_name != 'Bay Area 511 Regional VehiclePositions'
+)
+
+, deduped AS (
+    SELECT *,
+        {{ dbt_utils.surrogate_key(['dt', 'base64_url', 'location_timestamp', 'vehicle_id', 'vehicle_label', 'trip_id']) }} AS key,
+        {{ dbt_utils.surrogate_key(['dt', 'base64_url', 'vehicle_id', 'vehicle_label', 'trip_id']) }} AS path_key
```

Comment:
I do think we might want to call this like `vehicle_trip_key` or something like that, because `pathways` is another unrelated GTFS file

Reply:
🤦 fixing